src/charm.py

```diff
@@ -286,6 +287,37 @@ def _on_promote_user_admin_action(self, event: ActionEvent) -> None:
             return
         event.set_results(results)
 
+    def _on_anonymize_user_action(self, event: ActionEvent) -> None:
+        """Promote user admin and report action result.
+
+        Args:
+            event: Event triggering the promote user admin action.
+        """
+        results = {
+            "anonymize-user": False,
+        }
+        container = self.unit.get_container(synapse.SYNAPSE_CONTAINER_NAME)
+        if not container.can_connect():
+            event.fail("Failed to connect to the container")
+            return
+        try:
+            admin_access_token = self.get_admin_access_token()
+            if not admin_access_token:
+                event.fail("Failed to get admin access token")
+                return
+            username = event.params["username"]
+            admin = event.params["admin"]
+            server = self._charm_state.synapse_config.server_name
+            user = User(username=username, admin=admin)
+            synapse.deactivate_user(
+                user=user, server=server, admin_access_token=admin_access_token
+            )
+            results["anonymize-user"] = True
+        except synapse.APIError as exc:
+            event.fail(str(exc))
```

Comment:
It might be better to add some helpful messages here for UX than returning the exception stack :)